bindings/nodejs/src/functions/sol-platform-monitors.cc

```diff
@@ -0,0 +1,294 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <string.h>
+#include <nan.h>
+#include <sol-platform.h>
+
+#include "../hijack.h"
+#include "../common.h"
+#include "../structures/js-handle.h"
+
+using namespace v8;
+
+class SolPlatformMonitor: public JSHandle<SolPlatformMonitor> {
+public:
+	static const char *jsClassName() { return "SolPlatformMonitor"; }
+};
+
+#define ADD_MONITOR(name, marshaller) \
+	do { \
+    	VALIDATE_ARGUMENT_COUNT(info, 1); \
+    	VALIDATE_ARGUMENT_TYPE(info, 0, IsFunction); \
```

Comment:
Basically, all platform monitors except the service monitor have the same body. There's no point copying the same code over, like, eight times. I can't use a static function either, because each service monitor has a different signature, and I want to avoid casting them all to a common signature as C++ would require me to do. Besides, the ```info``` parameter is a variable whose type can vary from node version to node version, so I can't really pass it to a function. ```NAN_METHOD(some_binding)``` declares a function for me that accepts this parameter, but it never tells me its type.